exporters/autoexport/metrics.go

```diff
@@ -170,3 +221,31 @@ func getenv(key, fallback string) string {
 	}
 	return result
 }
+
+type producerRegistry struct {
+	envKey           string
+	fallbackProducer func(context.Context) (metric.Producer, error)
+	registry         *registry[metric.Producer]
+}
+
+func newProducerRegistry(envKey string) producerRegistry {
+	return producerRegistry{
+		envKey: envKey,
+		registry: &registry[metric.Producer]{
+			names: make(map[string]func(context.Context) (metric.Producer, error)),
+		},
+	}
+}
+
+func (pr producerRegistry) create(ctx context.Context) (metric.Producer, error) {
+	expType := os.Getenv(pr.envKey)
+	if expType == "" {
+		if pr.fallbackProducer != nil {
+			return pr.fallbackProducer(ctx)
+		}
+
+		return nil, nil
+	}
+
+	return pr.registry.load(ctx, expType)
```

Comment:
This currently loads a single producer from the registry, but the env var is plural: `OTEL_METRICS_PRODUCERS`.  Should we support multiple comma-separated values here?